src/app/courses/components/course-card/course-card.component.spec.ts

```diff
@@ -20,4 +31,73 @@ describe('CourseCardComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('is onCourseDelete method exists', () => {
+    const spy = spyOn(component, 'onCourseDelete');
+    component.onCourseDelete();
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('is onCourseEdit method exists', () => {
+    const spy = spyOn(component, 'onCourseEdit');
+    component.onCourseEdit();
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('when clicking on the edit button, the ID should be transmitted', () => {
+    component.course = course;
+    component.editCourse.subscribe(id => expect(id).toBe(course.id));
+    component.onCourseEdit();
+  });
+
```

Comment:
Unfortunately, this test do not test `onCourseEdit()`. You can check it. Comment out `component.onCourseEdit()` and run the test. The test will not fail.
To test the method you should create a spy for it first.
How to test this method I've described in the comment above.